test/src/main/java/org/apache/accumulo/test/functional/ServerSideErrorIT.java

```diff
@@ -63,69 +65,50 @@ public void run() throws Exception {
         bw.addMutation(m);
       }
 
-      boolean caught = false;
       // try to scan table
       try (Scanner scanner = c.createScanner(tableName, Authorizations.EMPTY)) {
-
-        try {
+        assertThrows(Exception.class, () -> {
           for (Entry<Key,Value> entry : scanner) {
             entry.getKey();
           }
-        } catch (Exception e) {
-          caught = true;
-        }
-
-        if (!caught)
-          throw new Exception("Scan did not fail");
-
-        // try to batch scan the table
-        try (BatchScanner bs = c.createBatchScanner(tableName, Authorizations.EMPTY, 2)) {
-          bs.setRanges(Collections.singleton(new Range()));
+        });
```

Comment:
Does this throw on `scanner.iterator()` or `iterator.next()`? I think this could get more specific on the assertThrows, both in the exception that is thrown and the method that is expected to throw, rather than wrap the whole loop.